### PR TITLE
Add docs interactive core functionality

### PR DESCRIPTION
**Pull request by michaellisitsa**
- [Add] Docs: Interactive diagram execution lifecycle, bundling and dev tools
- [Add] Docs: Interactive diagram iframe limitations

### CODE REVIEW

File: docs/interactive-diagram-rendering.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Interactive Diagram Rendering
+
+## Iframe sandbox
+
+Custom diagram should be restricted to use features available inside the `<iframe>` sandbox it is rendered in. This sandbox has following allowances `sandbox="allow-scripts allow-same-origin"`
```
[the remark this answers is not in the document]
Comment (michaellisitsa) [reply docs/interactive-diagram-rendering.md]: Good point. I've added further context 9cf88d2f239054570ce6c63ee8d0201d708056f5. It's a bit of a deep topic, but for the diagram creator, they don't have to think much more than they can use javascript and under-the-hood we message ClearCalcs.

File: docs/interactive-diagram-core.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+-   Bundling dependencies
+-   ESM Module Imports
+-   Sass [link](https://sass-lang.com/): Css stylesheets
+-   Support for Typescript
+-   Transpilation for older browser versions (i.e. last 2 years) - should not be changed
+-   Test Runner with hot reloading
+
+## Bundling
+
+On running `npm run compile`, [Parcel](https://parceljs.org/) generates a html file that compiles your code along with all bundled dependencies.
```
[the remark this answers is not in the document]
Comment (michaellisitsa) [reply docs/interactive-diagram-core.md]: They are [aliases](https://docs.npmjs.com/cli/v10/commands/npm-run-script), but for consistency updated in 92e71eaff92751be27815826e35cb561025491b7

File: docs/interactive-diagram-core.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Developer tooling
+
+The interactive custom diagram leverages modern developer tooling that can simplify the creation of complex diagrams with multiple files and dependencies whilst still compiling down to a html file that can be used inside of the user's browser. Much of this is enabled via the [Parcel](https://parceljs.org/) bundler. Features include:
+
+-   Code compilation and minification
+-   Asset importing e.g. images
+-   Bundling dependencies
+-   ESM Module Imports
+-   Sass [link](https://sass-lang.com/): Css stylesheets
```
[the remark this answers is not in the document]
Comment (michaellisitsa) [reply docs/interactive-diagram-core.md]: Updated in 92e71eaff92751be27815826e35cb561025491b7

File: docs/interactive-diagram-core.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Interactive Diagram Core
+
+## Execution Lifecycle
+
+Each interactive diagram has an execution lifecycle that is hidden away from the diagram creator. When a user navigates to a sheet with a custom diagram, the three phases begin with loading, initializing, then rendering.
+
+1. **loading**: HTML/CSS loaded and javascript executed (imported dependencies, top-level code)
+2. **initializing**: `initialize()` executed, await return/resolve.
+3. **rendering**: `render(params)` executed every time user changes params from sheet.
```
[the remark this answers is not in the document]
Comment (michaellisitsa) [reply docs/interactive-diagram-core.md]: Added links to code in 92e71eaff92751be27815826e35cb561025491b7. Don't want to go overboard with cross-linking.

**Comment by michaellisitsa** [comment]
@JulianGoh5 comments addressed

File: docs/interactive-diagram-core.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JulianGoh5) [reply docs/interactive-diagram-core.md]: Ohh good to know, thanks for the feedback

**Review by JulianGoh5 — APPROVED** [review]

**Pull request merged**